src/value.jl

```diff
@@ -1,145 +1,167 @@
-function CategoricalValue(level::Integer, pool::CategoricalPool{T, R}) where {T, R}
-    return CategoricalValue(convert(R, level), pool)
-end
-
-Base.convert(::Type{CategoricalValue{T, R}}, x::CategoricalValue{T, R}) where {T, R <: Integer} = x
-Base.convert(::Type{CategoricalValue{T}}, x::CategoricalValue{T}) where {T} = x
-Base.convert(::Type{CategoricalValue}, x::CategoricalValue) = x
-
-Base.convert(::Type{Union{CategoricalValue{T, R}, Null}}, x::CategoricalValue{T, R}) where {T, R <: Integer} = x
-Base.convert(::Type{Union{CategoricalValue{T}, Null}}, x::CategoricalValue{T}) where {T} = x
-Base.convert(::Type{Union{CategoricalValue, Null}}, x::CategoricalValue) = x
-
-Base.promote_rule(::Type{CategoricalValue{S, R}}, ::Type{T}) where {S, T, R} = promote_type(S, T)
-Base.promote_rule(::Type{CategoricalValue{S}}, ::Type{T}) where {S, T} = promote_type(S, T)
-Base.promote_rule(::Type{CategoricalValue}, ::Type{T}) where {T} = T
+# union of all types that have "categorical value" trait
+const CatValue{R} = Union{CategoricalValue{T, R} where T,
+                          CategoricalString{R}}
+
+# checks whether the type is "categorical value"
+iscatvalue(::Type) = false
+iscatvalue(::Type{Union{}}) = false # otherwise it dispatches to Type{<:...}
```

Comment:
I don't understand the comment. " dispatches to Type{<:...}" seems to imply it would dispatch to the method above, which sounds fine. Maybe this could be expanded a bit?

Reply:
Sorry, somehow I've missed that last round of comments. I will take a look shortly.